PositionalGuide/Plugin.cs

```diff
@@ -42,24 +44,41 @@ private enum CircleTypes { Target, Outer };
 
 	private readonly WindowSystem windowSystem;
 	private readonly ConfigWindow configWindow;
+	private readonly DtrBarEntry dtrEntry;
 
-	public Plugin() {
+	public Plugin(IDtrBar dtrBar) {
 		this.Config = Interface.GetPluginConfig() as Configuration ?? new();
 		this.Config.Update();
 
 		this.configWindow = new(this);
+		this.configWindow.OnSettingsUpdate += this.settingsUpdated;
```

Comment:
It _probably_ doesn't matter in this case, but in general, you're supposed to remove event handlers (use `-=`) on dispose. Just to be safe, I'd ask that you add that to the dispose method.

Reply:
The reason for removing handlers on dispose is that once an object is disposed, it should no longer be used, but event handlers have no way of knowing that and so will happily continue to call the method on the disposed object. Imagine if this method was a handler for something outside the plugin - when it tried to update the DTR bar entry, it would get an error because the entry was removed!

Reply:
Yeah make sense do not want dangling function pointers.